File: baba-chatbot-python/src/baba_chatbot/integrations/twilio/twilio_client.py
"""
Cliente para integración con Twilio WhatsApp API.
"""
import logging
from typing import Optional
from twilio.rest import Client


logger = logging.getLogger(__name__)


class TwilioClient:
    """
    Cliente para enviar mensajes a través de Twilio WhatsApp API.
    """

    def __init__(
        self,
        account_sid: str,
        auth_token: str,
        phone_number: str
    ):
        """
        Inicializa el cliente de Twilio.
        
        Args:
            account_sid: SID de la cuenta de Twilio
            auth_token: Token de autenticación de Twilio
            phone_number: Número de teléfono de Twilio (formato: whatsapp:+1234567890)
        """
        self._client = Client(account_sid, auth_token)
        self._phone_number = phone_number
        logger.info("TwilioClient initialized")

    def send_message(self, to: str, body: str) -> Optional[str]:
        """
        Envía un mensaje de WhatsApp a través de Twilio.
        
        Args:
            to: Número de destino (formato: whatsapp:+1234567890)
            body: Contenido del mensaje
            
        Returns:
            SID del mensaje enviado o None si hubo error
        """
        try:
            message = self._client.messages.create(
                from_=self._phone_number,
                body=body,
                to=to
            )
            logger.info(f"Message sent to {to}, SID: {message.sid}")
            return message.sid
        except Exception as ex:
            logger.error(f"Error sending message to {to}: {ex}")
            return None

File: baba-chatbot-net/src/Baba.Chatbot.Integrations/Twilio/TwilioClient.cs
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;
using Twilio.Rest.Api.V2010.Account;
using Twilio.Types;

namespace Baba.Chatbot.Integrations.Twilio;

/// <summary>
/// Cliente para enviar mensajes SMS/WhatsApp mediante la API de Twilio
/// </summary>
public class TwilioClient
{
    private readonly ILogger<TwilioClient> _logger;
    private readonly string _phoneNumber;

    /// <summary>
    /// Inicializa una nueva instancia del cliente de Twilio
    /// Configura las credenciales de autenticación
    /// </summary>
    public TwilioClient(IConfiguration configuration, ILogger<TwilioClient> logger)
    {
        _logger = logger;

        var accountSid = configuration["Twilio:AccountSid"] ?? throw new InvalidOperationException("Twilio AccountSid not configured");
        
        var authToken = configuration["Twilio:AuthToken"] ?? throw new InvalidOperationException("Twilio AuthToken not configured");

        _phoneNumber = configuration["Twilio:PhoneNumber"] ?? throw new InvalidOperationException("Twilio PhoneNumber not configured");

        global::Twilio.TwilioClient.Init(accountSid, authToken);
    }

    /// <summary>
    /// Envía un mensaje SMS/WhatsApp a un número de teléfono específico
    /// </summary>
    public async Task<string> SendMessageAsync(string to, string body)
    {
        try
        {
            var message = await MessageResource.CreateAsync(
                to: new PhoneNumber(to),
                from: new PhoneNumber(_phoneNumber),
                body: body
            );

            _logger.LogInformation("Message sent: {MessageSid} to {To}", message.Sid, to);
            return message.Sid;
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Failed to send message to {To}", to);
            throw;
        }
    }
}

